lib/net/ldap.rb

```diff
@@ -823,7 +829,11 @@ def search(args = {})
   #  ldap.port = 389
   #  ldap.auth your_user_name, your_user_password
   #  if ldap.bind
-  #    # authentication succeeded
+  #    if your_user_password.size > 0
```

Comment:
I think this complicates the example, should `auth`, `authenticate` or `bind` check for presence of the password internally when the authentication method is `simple`?.